### ISSUE TITLE
Passing size in the captcha request doesn't make any difference

### ISSUE THREAD

**Issue by kringo** [librecaptcha/lc-core#98]
Tried the following request with additional size map but didn't make any difference

{"level":"easy","media":"image/png","input_type":"text","size":{"width":100,"height":50}}

Is this an issue or i'm doing something wrong here?

**Comment by hrj** [comment]
I think we are not yet respecting the size parameter. Thanks for bringing it to our attention!

**Comment by kringo** [comment]
copy that, will wait for the update. Thx.

**Comment by kringo** [comment]
Wondering if this didn't make the cut in v1.1.0?

**Comment by hrj** [comment]
@kringo No, it didn't. We haven't been actively committing to the code recently, so I thought cutting out a release of the past unreleased commits would help move the project forward. Now that I am aware of [real world users](https://github.com/librecaptcha/lc-core/discussions/103), I might dedicate more time to this project.

---

### Design idea for this issue
* Define size buckets in the configuration file as an array of width and height. For example: `[ [200, 150], [300, 200], [600, 400] ]`
* When creating CAPTCHAs in background thread, create them for each defined size bucket.
* When serving CAPTCHAs, filter them by requested size.
* In future, an improvement to save resources could be: background thread only replenishes the size buckets which have been consumed.

If anyone would like to take this up, please drop a comment.

**Comment by hrj** [comment]
Just an update: I have started working on this. Images of the requested sizes are being created and served. The pending work is to scale the text to fit the requested size.

**Comment by kringo** [comment]
Thanks for the update @hrj, any thoughts to expect a release with these changes?

**Comment by hrj** [comment]
Sorry for the slow response; got busy with something else. I will hopefully get something out in April first week!

**Comment by kringo** [comment]
Thanks @hrj

**Comment by hrj** [comment]
@kringo Just merged the changes for this issue. Do give it a try.

Note:

* The API path is now changed to `/v2` because an extra parameter (size` is now required on the `/captcha` endpoint.
* The config section for each Captcha now has an additional setting: `allowedSizes`
* The Following Captcha Providers were modified to scale the text to fit the specified dimensions:
  * Shadow Text
  * Filter Challenge
  * Rain drops 
  * Popping characters
